crates/ruff/src/rules/pyflakes/fixes.rs

```diff
@@ -69,32 +60,131 @@ pub fn remove_unused_keyword_arguments_from_format_call(
     stylist: &Stylist,
 ) -> Result<Fix> {
     let module_text = locator.slice(location);
-    let mut tree = match_module(module_text)?;
-    let mut body = match_expr(&mut tree)?;
-
-    let new_call = {
-        let Expression::Call(call) = &body.value else {
-            bail!("Expected Expression::Call")
-        };
-
-        Call {
-            func: call.func.clone(),
-            lpar: call.lpar.clone(),
-            rpar: call.rpar.clone(),
-            whitespace_before_args: call.whitespace_before_args.clone(),
-            whitespace_after_func: call.whitespace_after_func.clone(),
-            args: call
-                .args
-                .iter()
-                .filter_map(|e| match &e.keyword {
-                    Some(kw) if unused_arguments.contains(&kw.value) => None,
-                    _ => Some(e.clone()),
-                })
-                .collect(),
-        }
+    let mut tree = match_expression(module_text)?;
+    let mut call = match_call(&mut tree)?;
+
+    call.args = call
+        .args
+        .iter()
+        .filter_map(|e| match &e.keyword {
+            Some(kw) if unused_arguments.contains(&kw.value) => None,
+            _ => Some(e.clone()),
+        })
+        .collect();
```

Comment:
Same as aboth: Can we use `Vec::retain` to "filter out" the unused arguments or `std::mem::take` with `into_iter` to avoid the `e.clone()`?